### ISSUE TITLE
Error starting Deepstack... 'redis-server.exe' processes did not start within

### ISSUE THREAD

**Issue by 162884** [VorlonCD/bi-aidetection#245]
@VorlonCD I'm running the latest beta version and I'm seeing this error. Any idea how to troubleshoot it? Thanks!

<img width="965" alt="Screen Shot 2021-05-12 at 8 46 06 PM" src="https://user-images.githubusercontent.com/38917259/118061772-9c478880-b363-11eb-836d-451cfaef65ee.png">

**Comment by VorlonCD** [comment]
Are you sure you are on the latest?   The last official release is really old.
https://github.com/VorlonCD/bi-aidetection/tree/master/src/UI/Installer

**Comment by 162884** [comment]
I believe so, I use the update function on the settings page to download/install the latest beta. Thanks!

**Comment by 162884** [comment]
It looks like I'm running the latest version of a tool and deep Stack CPU. Every time I try to stop and start deep Stack I get a status error.

![Screen Shot 2021-05-13 at 4 47 52 PM](https://user-images.githubusercontent.com/38917259/118186031-3b738b00-b40b-11eb-8fa5-47e0388b7ee7.png)
![Screen Shot 2021-05-13 at 4 49 21 PM](https://user-images.githubusercontent.com/38917259/118186055-43cbc600-b40b-11eb-9232-964e32f5978c.png)

**Comment by 162884** [comment]
@VorlonCD I unistalled/reinstalled DeepStack and the issue continues. Thanks!

**Comment by 162884** [comment]
@VorlonCD This seems like an odd error.

2021-05-16 10:20:08.880136|Info|AITOOLS.EXE|GetDeepStackRun|Deepstack_Objects:192.168.1.18:32785|Back_Porch_AI|None|  Deepstack partially running. Only 2 out of 2 copies of python.exe are running.|443|2||1

**Comment by VorlonCD** [comment]
Basically deepstack is expected to have 2 copies of python.exe running - Like this screenshot using [ProcessHacker:](https://processhacker.sourceforge.io/nightly.php)
![image](https://user-images.githubusercontent.com/7512113/118416142-3b2df680-b67c-11eb-9d05-2ff3c4e67210.png)

You can see that Deepstack.exe starts python.exe which should start a 3rd copy of python.exe.  Do you have 2 or less?

A few minutes after a failed start, hit STOP, then take a look at the errors in this file:
C:\Users\username\AppData\Local\DeepStack\logs\stderr.txt

You may be able to google for the error that shows up in there or post to the deepstack forums.

Did you try disabling realtime antivirus for a short time?

**Comment by 162884** [comment]
@VorlonCD I don't have the DeepStack folder excluded from anti-virus, should I?

**Comment by 162884** [comment]
@VorlonCD... I have 4 Python processes running but DeepStack shows an error.

   1 'redis-server.exe' processes did not start within 2390ms
<img width="965" alt="Screen Shot 2021-05-16 at 8 30 40 PM" src="https://user-images.githubusercontent.com/38917259/118418246-b2b45380-b685-11eb-9aea-c8a7a509c944.png">

**Comment by VorlonCD** [comment]
You shouldnt have 4 unless you have something else running that uses python.   Can you enable the COmmand Line column in Task Manager > Details, or ProcessHacker and send me a screenshot?

I still need the stderr.txt file contents.

**Comment by 162884** [comment]
Thanks, here is the screen shot from Task Manager that shows the commands for running each instance of Python. I checked my stderr.txt file and it's empty. Maybe logging is not enabled? If not, I'm not sure how to enable it.

![Screen Shot 2021-05-17 at 8 15 49 AM](https://user-images.githubusercontent.com/38917259/118487155-6d753d80-b6e8-11eb-80c9-9348a5121420.png)

**Comment by VorlonCD** [comment]
Looks like something else is starting another copy of deepstack.   If you have deepstack enabled in **BlueIris**, disable it since both cant be used at once.

If you dont know where it is starting from, use the ProcessHacker link above, and search for DeepStack to see which process started it in a tree view.

**Comment by 162884** [comment]
@VorlonCD I just wanted to let you know that I've made a few configuration changes that have helped.

I have two instances of DeepStack running... One on the Blue Iris PC and the other running on my QNAP in a Docker container. Previously, I had AI Tool automatically starting the DeepStack instance on the Blue Iris PC. I have now turned that off and set Blue Iris to control the starting of DeepStack. I also changed the setting for Blue Iris from using the Docker instance, to the local instance. AI Tool continues to have both instances configured for use.

After making these changes, the errors that I was seeing previously are not occurring anymore. I'm not sure why the configuration change made the difference, but I wanted you to know.
